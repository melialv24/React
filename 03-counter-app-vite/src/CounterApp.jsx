import PropTypes from "prop-types"
import { useState } from "react"

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState( value )
    const handleAdd = () => {
        setCounter(counter + 1)
    }

    return (
        <>
            <h1> CounterApp </h1>
            <h2> { counter } </h2>

            <button onClick={ handleAdd }>
                +1
            </button>
        </>
    )
}

CounterApp.PropTypes = {
    value: PropTypes.number.isRequired
}
