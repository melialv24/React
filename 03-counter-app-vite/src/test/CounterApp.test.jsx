import {render} from '@testing-library/react'
import { CounterApp } from "../CounterApp"


describe('Prueba del counter app', () => {

    const value = 10

    test('Debe de hacer match con el snapchot', () => {

        const { container } = render(<CounterApp value = { value } />)
        expect( container ).toMatchSnapshot()
    })

    test('Debe de mostrar el valor inicial de 100', () => {
        const { container, getByText } = render(<CounterApp value = { value } />)
        expect( getByText( value )).toBeTruthy()
    })
})