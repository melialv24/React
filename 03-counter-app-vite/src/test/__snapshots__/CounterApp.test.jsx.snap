// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del counter app Debe de hacer match con el snapchot 1`] = `
<div>
  <h1>
     CounterApp 
  </h1>
  <h2>
     
    10
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
